### PR TITLE
Fixed utilsTest/sortTest and serviceTest (resolves #272)

### LINKED ISSUE #272: Service-based jobs need user script resolution
Continued from #268

### PR DESCRIPTION
**Pull request by benedictpaten**
was set to equal 1. This is fixed, as is the serviceTestIssue. Hannes,
please test this now works for you on pyCharm.

### CODE REVIEW

**Comment by hannes-ucsc** [comment]
JobServiceTest works on PyCharm. A call to ModuleDescriptor.globalize() was missing, which I've just added.

It just struck me that we probably don't need the "scriptTree" dummy command name.

**Comment by hannes-ucsc** [comment]
BTW, the commit message of the commit above doesn't really serve its purpose. Procedural notes, instructions and observations and should go into the issue or PR comments. Once the PR is merged, all of that contextual information in the comments of the underlying issue and the PR conversation will be detached from the Git history. When that happens, this particular commit message will not make much sense to someone reading through the history in a few weeks or months.

tl;dr the commit message is the most important vehicle to communicate the intent of a commit. It's also hardest to fix after the fact.

**Comment by benedictpaten** [comment]
Point taken, I will do a better job in future.

**Comment by hannes-ucsc** [comment]
Jenkins, test this please.

**Comment by benedictpaten** [comment]
Just pulled in your changes, master and a couple of comment changes. It produced some conflicts, which I think I resolved properly...

**Comment by hannes-ucsc** [comment]
Actually, it looks like your merge killed my changes. Unfortunately, I realized that after I merged. I will have to rollback the merge which will be rewriting history.

**Comment by hannes-ucsc** [comment]
Rolled back master, restored my changes, cherry-picked your "Added some comments Job._loadClass", squashed and rebased on top of the rolled back master. Filed new PR #289.

**Comment by benedictpaten** [comment]
Oh damn, sorry, I thought the merge was a bit too easy, thanks for fixing.

Benedict
(from phone)
